### ISSUE TITLE
Create tunnels

### ISSUE THREAD

**Issue by t0stiman** [Insprill/dv-mapify#51]
Add the ability to create tracks with tunnels, like the ones in the base game or with a custom prefab.

**Comment by Kelticcol** [comment]
https://assetstore.unity.com/packages/tools/terrain/digger-terrain-caves-overhangs-135178

The only unfortunate thing..... not compatible with the 2019.4.40 version we use..........

**Comment by Kelticcol** [comment]
Since i have requested help on a few issues, i thought this would be appreciated.

I have made a tunnel, yes, its only a straight and short one, but it works, might look a little dull, but hey beggars can't be choosers right?

![Tunnel 1](https://github.com/Insprill/dv-mapify/assets/152837201/e58520be-b0d2-4af5-a0fe-a35718d79eba)
![Tunnel 1b](https://github.com/Insprill/dv-mapify/assets/152837201/752ec973-fc13-4fd7-9813-781f75a4902e)

If you happen to be interested in how i made said tunnel i will be happy to share.
Also if i can figure out how to package it id send it too.

**Comment by WhistleWiz** [comment]
Can a train pass through them?

**Comment by Kelticcol** [comment]
Oh yes they can quite well too.
The only slight issue at the moment is that the interior terrain has no mesh colliders, so if you walk through the tunnel stay on the tracks.

**Comment by WhistleWiz** [comment]
Not sure how DV does it but you could try placing box colliders along the edges (automatically of course).
I think DV simply has the holes in the terrain then extrudes a mesh along the track to make the tunnel, but don't quote me on that.

**Comment by Kelticcol** [comment]
Yeah I'm adding in box colliders, 2d for the walls and 3d for the roof and floor, will let you know how it goes.

**Comment by Kelticcol** [comment]
Tunnel works fine, 3d colliders for roof, floor and walls.

I do have a 150mb clip of the de6 with a load going through the tunnel, any idea how to upload?

**Comment by t0stiman** [comment]
Upload to Youtube, make it unlisted, post a link here

**Comment by Insprill** [comment]
I've had this on the project board since April, I just forgot to turn it into an issue. The ideal implementation would allow you to simply run the track as you'd like and any parts of it that are fully underground would automatically be turned into tunnels.

**Comment by t0stiman** [comment]
I was going to ask them to share their project, but they deleted their GitHub account...

**Comment by WhistleWiz** [comment]
The only issue is making sure the train goes through the terrain holes. DV has a bunch of colliders at the entrances too for a bunch of things (for example, interacting with steamer smoke), the tunnel walls is just a simple mesh sweep.

Colliders inside can be placed by sweeping multiple boxes too, just like the track colliders (multiple boxes to create the gravel shape).
